### PR TITLE
Add Accordion titles and heading levels

### PR DESCRIPTION
**Pull request by tqureshi-uog**
# Summary of changes
Content editors now have the ability to add a title for an `Accordion` widget and set its heading level. They can also set the heading level for each `Accordion` item title.

## Frontend
- Add new fields to `gatsby-node.js` schema
- Add accordion title and heading level to `accordion.js`
- Add accordion item heading levels to `accordion.js`

### Incidental changes
- Reduce top margin on `Section` widget title in `widget.js`
- Upgrade packages (keeps this repo in sync with `doorknob`)

## Backend
- Add optional title field and heading level to `Accordion` paragraph type
- Add optional heading level field to `Accordion Item` title
- Update relevant field labels and instructions

# Test Plan

Gatsby Cloud site: https://tqtest.gatsbyjs.io/accordion-test/
Drupal multidev: https://accordionh-chug.pantheonsite.io/
PR build: https://ugconthub-accordionh.gatsbyjs.io/

1. Go to https://tqtest.gatsbyjs.io/accordion-test/ and confirm all headings are nested properly
2. Compare https://tqtest.gatsbyjs.io/international/admission-requirements/ to https://admission.uoguelph.ca/international/admission-requirements/ - the former should have a proper and logical heading structure. See https://accordionh-chug.pantheonsite.io/node/997/edit for how it was done using the new accordion heading features 
3. For a quick and convenient method of seeing the heading structure of a page, install the [HeadingsMap Chrome plugin](https://chrome.google.com/webstore/detail/headingsmap/flbjommegcjonpdmenkdiocclhjacmbi) (also available for Firefox)

**Note:** For the moment, `field_heading_level` offers `h2`, `h3`, and `h4` as choices with `h2` being the default. Options `h5` and `h6` are omitted since it's difficult to imagine a legitimate use case for them.

## Known Issues
The default `h2` setting for `field_heading_level` will only show up when creating new `Accordion` widgets. On existing `Accordion` widgets (i.e. created before `field_heading_level` was available), the setting will show as "None" (if left like this, it will render as `h2` in the Gatsby frontend component)

### CODE REVIEW

**Comment by abanuog** [comment]
For the heading level dropdown, can you set the default to h2 (instead of none), and remove none, since it's not really an option ?

**Comment by tqureshi-uog** [comment]
> For the heading level dropdown, can you set the default to h2 (instead of none), and remove none, since it's not really an option ?

The default is already set to h2:

image

...but it will only take effect for new content (you can test this by going to https://accordionh-chug.pantheonsite.io/node/add/page and adding an `Accordion` widget). On existing accordion widgets, the setting will default to `None`

File: src/components/shared/accordion.js [hunk 1]
```diff
@@ -1,29 +1,67 @@
...
+                  data-bs-toggle="collapse"
+                  data-bs-target={"#part" + item.drupal_id}
+                  aria-expanded="false"
+                  aria-controls={"part" + item.drupal_id}
+                >
+                  {item?.field_accordion_title ? item.field_accordion_title : "Read More"}
+                </button>
+              );
+
+              let headingElement;
```
Comment (mmafe) [inline src/components/shared/accordion.js:31]: Try replacing lines 31 through 53 with the following:

```
              let HeadingLevelItem = (
                (item?.field_heading_level === "h3" || item?.field_heading_level === "h4") 
                ? item.field_heading_level : "h2");

              let headingElement = (
                <HeadingLevelItem className="accordion-header" id={"heading" + item.drupal_id}>
                  {accordionToggle}
                </HeadingLevelItem>
              )
``` #

**Review by mmafe — COMMENTED** [review]
Added a code snippet for shortening one segment. 

Other than that, nesting works like a dream.

Only thought is that occasionally it looks like users have been adding a class <text-dark> to make the heading above the accordion dark. That will no longer be an option for users, however I don't think it should be an option (and the h3s are going to be dark anyway in future, so that'll eventually sort itself out.)

File: src/components/shared/accordion.js (hunk 1 above)
Comment (mmafe) [reply src/components/shared/accordion.js]: This comment is resolved with the latest update.

**Review by mmafe — CHANGES_REQUESTED** [review]
As discussed on Teams, we are adding an h5 and h6 option to the Accordion Item on the Drupal side. Will review one more time after that occurs.

**Comment by tqureshi-uog** [comment]
> As discussed on Teams, we are adding an h5 and h6 option to the Accordion Item on the Drupal side. Will review one more time after that occurs.

Done.

**Review by mmafe — APPROVED** [review]

**Review by snschau — APPROVED** [review]

**Pull request merged**